src/components/side_panel/chart/chart_with_axis/design_panel.ts

```diff
@@ -210,6 +215,11 @@ export class ChartWithAxisDesignPanel<P extends Props = Props> extends Component
       element.value = `${this.getTrendLineConfiguration()?.order ?? 2}`;
       return;
     }
+    const maxOrder = this.getMaxPolynomialDegree();
+    if (order > maxOrder) {
+      element.value = `${this.getTrendLineConfiguration()?.order ?? maxOrder}`;
```

Comment:
Is this still needed since it's a selection input?  (the user can't enter an arbitrary value)